src/assisted_installer_controller/assisted_installer_controller_test.go

```diff
@@ -1098,7 +1098,27 @@ var _ = Describe("installer HostRoleMaster role", func() {
 				assistedController.UpdateNodeLabels(context.TODO(), &wg)
 				wg.Wait()
 			})
+			It("Set label only in Ready node", func() {
+				nodeLabels := `{"node.ocs.openshift.io/storage":""}`
+				k8sNodes := GetKubeNodes(kubeNamesIds)
+				for i, cond := range k8sNodes.Items[0].Status.Conditions {
+					if cond.Type == v1.NodeReady {
+						k8sNodes.Items[0].Status.Conditions[i].Status = v1.ConditionFalse
+					}
+				}
+
+				hosts := create3Hosts(models.HostStatusInstalled, models.HostStageDone, nodeLabels)
+				mockbmclient.EXPECT().GetHosts(gomock.Any(), gomock.Any(), []string{models.HostStatusDisabled, models.HostStatusError}).
+					Return(hosts, nil).Times(2)
+				mockk8sclient.EXPECT().ListNodes().Return(k8sNodes, nil).Times(1)
+				mockk8sclient.EXPECT().ListNodes().Return(GetKubeNodes(kubeNamesIds), nil).Times(1)
+				mockk8sclient.EXPECT().PatchNodeLabels(gomock.Any(), nodeLabels).Return(nil).Times(2)
+				mockk8sclient.EXPECT().PatchNodeLabels(gomock.Any(), nodeLabels).Return(nil).Times(1)
```

Comment:
I don't get it, in the first call  `ListNodes` return not ready nodes, in the second time it get called  `ListNodes` return ready nodes.
Why `PatchNodeLabels` get called 3 times?

Reply:
it runs in a loop till all nodes get there labels.
1. Set labels of 2 hosts as they are ready and don't set for the third one as it is not
2. Set labels of the last host that we set ready

Reply:
Hmmm, I don't understand.
you have 3 nodes, 2 of them should be skipped due the the status in AI ([here](https://github.com/openshift/assisted-installer/pull/620#discussion_r1089900370))
That leaves one node, what am I missing?

Reply:
See the function and all other tests, this test doesn't validate any status in assisted and calls GetHosts with statuses exactly the same as function do